org.civicrm.reqangular/src/common/directives/time-amount-picker.directive.js

```diff
@@ -96,17 +97,21 @@ define([
      * Calculates the output number in hours (float) and sets to the scope
      */
     function calculateSelectedValue () {
-      if (vm.selectedHours === '' || vm.selectedMinutes === '') {
-        return null;
+      if (vm.selectedHours !== '' && vm.selectedMinutes !== '') {
+        $scope.value = +vm.selectedHours + vm.selectedMinutes / 60;
       }
-
-      $scope.value = +vm.selectedHours + vm.selectedMinutes / 60;
     }
 
     /**
      * Starts watching inbound attributes for changes
      */
     function watchTimeAmountPickerOptions () {
+      $scope.$watch('disabled', function (disabled) {
```

Comment:
This $scope.$watch is not needed since vm.disabled will be automatically updated

Reply:
@reneolivo vm.disabled will not be automatically updates since scope binding are not bound to the controller.